app.go

```diff
@@ -461,6 +473,20 @@ func (app *App) StopTimeout() time.Duration {
 	return app.stopTimeout
 }
 
+// DotGraph provides a graph of dependencies in the container as a string in DOT
+// syntax.
+func (app *App) DotGraph() DotGraph {
+	// Memoize the graph
+	if app.dotgraph != nil {
```

Comment:
super nit: you can invert the if statement to deal with creation and then always return the dotgraph

Reply:
I think this is actually clearer logically, even if it is not optimized for fewest return statements.